.github/workflows/integration_test.yml

```diff
@@ -1,43 +1,66 @@
-    - uses: conda-incubator/setup-miniconda@v2.0.0
+    -
+      name: Log in to the Container registry
+      uses: docker/login-action@v1
       with:
-        miniconda-version: 'latest'
-        activate-environment: test-environment
-        python-version: 3.6.10
-        auto-activate-base: false
-        auto-update-conda: true
-        condarc-file: test/condarc.yml
-
-    - name: Use Node JS 16.x
-      uses: actions/setup-node@v1
+        registry: ghcr.io
+        username: ${{ secrets.GHCR_USERNAME }}
+        password: ${{ secrets.GHCR_TOKEN }}
+    -
+      name: Repo checkout
+      uses: actions/checkout@v2
+    -
+      name: Set up environment
+      run: |
+        if [ ${{ github.base_ref }} == 'master' ]; then
+            branch=''
+        else
+            branch="-""${{ github.base_ref }}"
+        fi
+        narrative_version=`grep '\"version\":' src/config.json.templ | awk '{print $2}' | sed 's/"//g'`
+        narrative_git_hash=`grep '\"git_commit_hash\":' src/config.json.templ | awk '{print $2}' | sed 's/"//g' | sed 's/,//'`
+        echo "APP_IMAGE_TAG=ghcr.io/${{ github.repository }}${branch}:pr-${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+        echo "APP_VERSION_IMAGE_TAG=ghcr.io/${{ github.repository }}${branch}_version:pr-${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+        echo "BRANCH=${branch}" >> $GITHUB_ENV
+        echo "BUILD_DATE=$(date -u +"%Y-%m-%dT%H:%M:%SZ")" >> $GITHUB_ENV
+        echo "NARRATIVE_GIT_HASH=${narrative_git_hash}" >> $GITHUB_ENV
+        echo "NARRATIVE_VERSION=${narrative_version}" >> $GITHUB_ENV
+        echo "PR=pr-${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+        echo "VCS_REF=$(echo "${{ github.event.pull_request.head.sha }}" | cut -c -7)" >> $GITHUB_ENV
+    -
+      name: Pull docker image
+      shell: bash -l {0}
+      run: docker pull ${{ env.APP_IMAGE_TAG }}
+    -
+      name: Use Node JS 16.x
+      uses: actions/setup-node@v2
       with:
         node-version: 16.x
-
-    - name: Install JS dependencies
+    -
+      name: Install JS dependencies
       run: |
         npm ci
         npm run install-npm
-
-    - name: Install Narrative Application
-      shell: bash -l {0}
-      run: |
-          bash ./scripts/install_narrative.sh
-          npm run minify
-          sed <src/config.json.templ >src/config.json "s/{{ .Env.CONFIG_ENV }}/dev/"
-          sed -i 's/{{ if ne .Env.CONFIG_ENV "prod" }} true {{- else }} false {{- end }}/true/' src/config.json
-          jupyter notebook --version
-
-    - name: Run Narrative Frontend Integration Tests
+        cp src/config.json kbase-extension/static/kbase/config/
+    -
+      name: Run Narrative Frontend Integration Tests
       shell: bash -l {0}
       env:
+        BASE_URL: http://localhost:32323
         KBASE_TEST_TOKEN: ${{ secrets.NARRATIVE_TEST_TOKEN }}
-      run: make test-integration
+      run: bash scripts/run_tests.sh -i -c ${{ env.APP_IMAGE_TAG }}
```

Comment:
run integration tests against the docker image built in the `build_and_push_image.yml` workflow